### PR TITLE
scripts: unity: Fix header_prepare comments removal regex

### PR DESCRIPTION
**Pull request by nordic-krch**
scripts: unity: Fix header_prepare comments removal regex

Regex for comments removal was also removing strings which resulted in corrupted files if includes were in quotation marks (`#include "foo.h"`).

tests: unity: example_test: Add various comments to mock header

In order to test that header_prepare.py script is properly handling comments removal from the mocked header, added various types of comments to foo.h.

### CODE REVIEW

**Comment by evenl** [comment]
I have a huge set of unity tests that I am porting to the NCS build system. I have ported some tests that validate that this update works for my tests at least.

**Comment by nordic-krch** [comment]
@tejlmand can you take another look?

File: scripts/unity/header_prepare.py [hunk 1]
```diff
@@ -13,19 +13,26 @@ def header_prepare(in_file, out_file, out_wrap_file):
...
+    c_comments_pattern = re.compile(
+        r'/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/',
         re.DOTALL | re.MULTILINE
     )
-    content = comments_pattern.sub(r"", content)
+
+    content = c_comments_pattern.sub(r"", content)
+
+    cpp_comments_pattern = re.compile(
+        r'(?!\".?)\/\/\w+$',
```
Comment (tejlmand) [inline scripts/unity/header_prepare.py:24]: The `$` in this case is not working as expected:
```
with open('test.txt') as f_in:
    content = f_in.read()
content = c_comments_pattern.sub(r"", content)
content = cpp_comments_pattern.sub(r"", content)

print(content)
...
#include "foo_internal.h"
// c++ comment
...
```

This alse means that comments following code, such as:
```
int foo_init(void *handle); // foo_init declaration
```
are also not removed correctly.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please check the `cpp_comments_pattern` regex.

File: scripts/unity/header_prepare.py (hunk 1 above)
Comment (nordic-krch) [reply scripts/unity/header_prepare.py]: right, can you take another look? Changed to:

```
cpp_comments_pattern = re.compile(
        r'(?!.*\".*)\/\/.*$',
        re.DOTALL | re.MULTILINE
    )
```

Comment (tejlmand) [reply scripts/unity/header_prepare.py]: Still not working when loading a file into python shell, and testing it there.
Wonder if the `re.MULTILINE` could be impacting the `$` in the regex in this case.

Comment (nordic-krch) [reply scripts/unity/header_prepare.py]: removed `re.DOTALL` and that helped. Please check again.

Comment (tejlmand) [reply scripts/unity/header_prepare.py]: It sure did :+1:

**Review by tejlmand — APPROVED** [review]
lgtm

**Comment by nordic-krch** [comment]
@rlubos @joerchan can you merge?

**Pull request merged**